### ISSUE TITLE
Previewkernel expects an App\Kernel

### ISSUE THREAD

**Issue by TimGroenenberg** [sulu/sulu#4782]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | no
| Sulu Version | 2.0.0-RC3
| Browser Version | Browser name and version

#### Actual Behavior

The `PreviewKernel` extends the `App/Kernel` [here](https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewKernel.php#L14). This means you can't change the namespace of your `src/Kernel` without adding an alias for this class.

#### Expected Behavior

I expect I should be able to change the namespace of my project `src` dir without breaking the preview.

#### Steps to Reproduce

- Create a new project based on [sulu/skeleton](https://github.com/sulu/skeleton)
- Change the namespace of your `src` dir and in the `Kernel` file.
- Edit a page in Sulu and an error will be thrown in the preview.

**Comment by alexander-schranz** [comment]
@TimGroenenberg sorry this is currently a limitation. You could create a [class_alias](https://www.php.net/manual/de/function.class-alias.php) or create a `App\Kernel` which extends from your Kernel.

**Comment by alexander-schranz** [comment]
@TimGroenenberg another way would you overwrite the `sulu_preview.preview.kernel_factory` service to create the a custom PreviewKernel for your project.
If you want you can create a PR that there is a configuration paramer for the `PreviewKernel` class and we provide the logic of it as a trait so you can do:

```php
class CustomPreviewKernel extends CustomKernelClass {
     use PreviewKernelTrait;
}
```

**Comment by alexander-schranz** [comment]
Seems like there exist an old issue here #2601 with a suggestion how we could solve it.

**Comment by danrot** [comment]
Yeah, but the old issue references the old structure... So I am going to close the old one, since it is referenced here anyway :slightly_smiling_face:
